pkgs/development/haskell-modules/lib.nix

```diff
@@ -193,6 +193,42 @@ rec {
 
   disableHardening = drv: flags: overrideCabal drv (drv: { hardeningDisable = flags; });
 
+  /* Register the bash completion scripts that are provided by
+   * optparse-applicative. The resulting package will have a
+   * share/bash-completion/completions/${command} file.
+   *
+   *   appendCompletion command pkg
+   *
+   *     command: name of an executable
+   *         pkg: Haskell package that build the executable
+   */
+  appendCompletion = command: pkg:
```

Comment:
I believe that `generateCompletion` is a more descriptive name than `appendCompletion`. Also, there might be more than one way to generate those files depending on which option parsing library is used, therefore it might be a good idea to put the `optparse-applicative` in there somehow, i.e. `generateOptparseApplicativeCompletion`.